content/src/main/content/jcr_root/apps/acs-commons/touchui-widgets/showhidefields/source/checkboxshowhide.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Extension to the standard checkbox component. It enabled hidding/unhidding of multiple dialog fields based on the
+ * toggling of checkbox. 
+ *
+ * - add the class cq-dialog-checkbox-showhide to the checkbox element
+ * - add the data attribute cq-dialog-checkbox-showhide-target to the checkbox element, value should be the
+ *   selector, usually a specific class name, to find all possible target elements that can be shown/hidden.
+ * - add the target class to each target component that can be shown/hidden
+ * - add the class hidden to each target component to make them initially hidden
+ * - add the attribute showhidetargetvalue to each target component, the value should equal to:
+ * 	 'true', if the field is to be displayed when Checkbox is selected.
+ * 	 'false', if the field is to be displayed when Checkbox is unselected.
+ */
+(function(document, $) {
+    "use strict";
+
+    // when dialog gets injected
+    $(document).on("foundation-contentloaded", function(e) {
+        // if there is already an inital value make sure the according target element becomes visible
+        showHide($(".cq-dialog-checkbox-showhide", e.target));
```

Comment:
@davidjgonzalez : I have generalized the attribute to set target "cq-dialog-showhide-target". 

Can I please request you to reconsider the generalization of  input field selectors "[data-cq-dialog-dropdown-showhide], [data-cq-dialog-checkbox-showhide]".
Reason: The selectors are being reused for registering events as well.
- selected for dropdown
- checked for checkbox.